src/Core/Translation/Provider/ThemeCatalogueLayersProvider.php

```diff
@@ -0,0 +1,236 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Translation\Provider;
+
+use PrestaShop\PrestaShop\Core\Addon\Theme\Theme;
+use PrestaShop\PrestaShop\Core\Addon\Theme\ThemeRepository;
+use PrestaShop\PrestaShop\Core\Exception\FileNotFoundException;
+use PrestaShopBundle\Translation\Extractor\ThemeExtractor;
+use PrestaShopBundle\Translation\Loader\DatabaseTranslationLoader;
+use PrestaShopException;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Translation\MessageCatalogue;
+
+/**
+ * Returns the 3 layers of translation catalogues related to the Backoffice interface translations.
+ * The default catalogue is in app/Resources/translations/default, in any file starting with "Admin"
+ * The file catalogue is in app/Resources/translations/LOCALE, in any file starting with "Admin"
+ * The user catalogue is stored in DB, domain starting with Admin and theme is NULL.
+ *
+ * @see CatalogueLayersProviderInterface to understand the 3 layers.
+ */
+class ThemeCatalogueLayersProvider implements CatalogueLayersProviderInterface
+{
+    /**
+     * We need a connection to DB to load user translated catalogue.
+     *
+     * @var DatabaseTranslationLoader
+     */
+    private $databaseTranslationLoader;
+    /**
+     * @var string
+     */
+    private $themeName;
+    /**
+     * @var CatalogueLayersProviderInterface
+     */
+    private $coreFrontProvider;
+    /**
+     * @var ThemeExtractor
+     */
+    private $themeExtractor;
+    /**
+     * @var ThemeRepository
+     */
+    private $themeRepository;
+    /**
+     * @var Filesystem
+     */
+    private $filesystem;
+    /**
+     * @var string
+     */
+    private $themeResourcesDir;
+    /**
+     * @var \PrestaShop\PrestaShop\Core\Addon\Theme\Theme
+     */
+    private $theme;
```

Comment:
```suggestion
    /**
     * We need a connection to DB to load user translated catalogue.
     *
     * @var DatabaseTranslationLoader
     */
    private $databaseTranslationLoader;

    /**
     * @var string
     */
    private $themeName;

    /**
     * @var CatalogueLayersProviderInterface
     */
    private $coreFrontProvider;

    /**
     * @var ThemeExtractor
     */
    private $themeExtractor;

    /**
     * @var ThemeRepository
     */
    private $themeRepository;

    /**
     * @var Filesystem
     */
    private $filesystem;

    /**
     * @var string
     */
    private $themeResourcesDir;

    /**
     * @var \PrestaShop\PrestaShop\Core\Addon\Theme\Theme
     */
    private $theme;
```

Let this poor properties breath :sweat_smile: